components/brave_webtorrent/browser/net/brave_torrent_redirect_network_delegate_helper.cc

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_webtorrent/browser/net/brave_torrent_redirect_network_delegate_helper.h"
+
+#include "base/strings/strcat.h"
+#include "base/strings/string_util.h"
+#include "brave/common/extensions/extension_constants.h"
+#include "extensions/common/constants.h"
+#include "net/http/http_content_disposition.h"
+#include "net/http/http_response_headers.h"
+#include "net/url_request/url_request.h"
+
+namespace {
+
+bool FileNameMatched(const net::HttpResponseHeaders* headers) {
+  std::string disposition;
+  if (!headers->GetNormalizedHeader("content-disposition", &disposition)) {
+    return false;
+  }
+
+  net::HttpContentDisposition cd_headers(disposition, std::string());
```

Comment:
++ on using `HttpContentDisposition` since the header can be complex like
`Content-Disposition: attachment; filename=genome.jpeg;   modification-date="Wed, 12 Feb 1997 16:29:51 -0500";`